### PR TITLE
Cleanup sound manager class

### PR DESCRIPTION
**Pull request by nerzhul**
* Use some const refs
* Use auto on iterators
* Drop unused parameters
* move sound managers to client folder & cmakelists
* don't include sound libraries for every module, only for client

### CODE REVIEW

**Comment by nerzhul** [comment]
This is a sound refactor/reorganization PR. I will work to make it event-based after this refactor to make client less lag-prone when loading files and GUI runs

File: src/client/sound.h [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+public:
+	virtual bool loadSoundFile(const std::string &name, const std::string &filepath)
+	{
+		return true;
+	}
+	virtual bool loadSoundData(const std::string &name, const std::string &filedata)
+	{
+		return true;
+	}
+	void updateListener(const v3f &pos, const v3f &vel, const v3f &at, const v3f &up) {}
```
Comment (Dumbeldor) [inline src/client/sound.h]: you can used "override" or "final" ;)

Comment (nerzhul) [reply src/client/sound.h]: i can, it's just a port of the original version

Comment (Dumbeldor) [reply src/client/sound.h]: Okay, you can improve the original version :+1:

**Comment by nerzhul** [comment]
@SmallJoker when you get time, please review :)

File: src/client/CMakeLists.txt [hunk 2]
```diff
@@ -1,4 +1,20 @@
+if(USE_SOUND)
+	set(sound_SRCS
+		${CMAKE_CURRENT_SOURCE_DIR}/sound_openal.cpp
+		${CMAKE_CURRENT_SOURCE_DIR}/sound.cpp)
```
Comment (SmallJoker) [inline src/client/CMakeLists.txt]: `sound.cpp` must be included regardless on `USE_SOUND` to have the DummySoundManager available.

Comment (nerzhul) [reply src/client/CMakeLists.txt]: exact, fixing that

**Comment by nerzhul** [comment]
Approved by @SmallJoker (http://irc.minetest.net/minetest-dev/2018-03-24#i_5258402)

**Pull request merged**